src/data/finset/noncomm_prod.lean

```diff
@@ -115,6 +115,12 @@ on all elements `x ∈ s`." ]
 def noncomm_prod (s : multiset α) (comm : ∀ (x ∈ s) (y ∈ s), commute x y) : α :=
 s.noncomm_fold (*) comm 1
 
+/-- A helper for rewriting the dependent argument. -/
+@[to_additive "A helper for rewriting the dependent argument "]
+lemma noncomm_prod_congr
+  {s₁ s₂ : multiset α} (h : s₁ = s₂) (comm : ∀ (x ∈ s₁) (y ∈ s₁), commute x y) :
+  noncomm_prod s₁ comm = noncomm_prod s₂ (h ▸ comm) := by subst h
+
```

Comment:
I don't think we really need this lemma; `simp_rw h`, works just fine, and we don't have `multiset.prod_congr` either.

Reply:
`multiset.prod_cong` doesn't have a dependent argument, which gets in the way of rewriting. Or maybe I'm doing it wrong, and we don't need any of these lemmas?

Reply:
> `multiset.prod` doesn't have a dependent argument, which gets in the way of rewriting.

True; but we have a general solution to "`rw` doesn't work because of a dependent argument", it's `simp_rw`. Creating a congruence lemma for every dependent term is just manually doing what the mechanism behind `simp` already does.

Note that `finset.non_comm_prod_congr` _is_ useful, because the automatic congruence lemma doesn't know that f and g only need to be equal within the set.

Reply:
```suggestion
```